Sources/AppcuesKit/Presentation/Actions/Appcues/AppcuesSubmitFormAction.swift

```diff
@@ -16,14 +16,20 @@ internal class AppcuesSubmitFormAction: AppcuesExperienceAction, ExperienceActio
 
     static let type = "@appcues/submit-form"
 
+    private weak var appcues: Appcues?
+
     let skipValidation: Bool
 
     required init?(configuration: AppcuesExperiencePluginConfiguration) {
+        self.appcues = configuration.appcues
+
         let config = configuration.decode(Config.self)
         self.skipValidation = config?.skipValidation ?? false
     }
 
-    func execute(inContext appcues: Appcues, completion: ActionRegistry.Completion) {
+    func execute(completion: ActionRegistry.Completion) {
+        guard let appcues = appcues else { return completion() }
```

Comment:
in this case would you want to have this guard after the `defer` on line below and then just have the completion call in one spot?

Reply:
Definitely an option. I did it like this for the sake of consistency with all the other actions. Happy to update it if you think it'd be better.

Reply:
yeah, argument for consistency makes sense, sounds good to me.